# Erwan LeCun
# 8/5/2015
# PUI HW 3-2 

In [3]:
import os
import sys
import numpy as np
import pylab as pl

alpha = 0.05

P_0 = 52.1 * 0.01
P_1 = 53.3 * 0.01

n_0 = 409
n_1 = 564

Nt_0 = P_0 * n_0
Nt_1 = P_1 * n_1

sp = (P_0 * n_0 + P_1 * n_1) / (n_1 + n_0)
sp_stdev = lambda p, n: np.sqrt(p * (1 - p) / n[0] + p * (1 - p) / n[1])
sp_stdev_3y = sp_stdev((Nt_0 + Nt_1) / (n_0 + n_1), [n_0, n_1])
zscore = lambda p0, p1, s: (p0 - p1) / s
z_3y = zscore(P_1, P_0, sp_stdev_3y)
p_3y = 1 - 0.6443

print z_3y

def report_result(p, a):
    print 'is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(p, a)
    if p < a:
        print "YES!"
    else:
        print "NO!"
    print 'the Null hypothesis is {}'.format('rejected' if p < a else 'not rejected')

report_result(p_3y, alpha)

0.370114343926
is the p value 0.36 smaller than the critical value 0.05? 
NO!
the Null hypothesis is not rejected


In [4]:
Ntot = 973
expected = 564 * 409 * 131.579 * 841.421
sample_values = [[14.7 * 5.64, 85.3 * 5.64], [11.9 * 4.09, 88.1 * 4.09]]
chisqstat = lambda N, values, expect: N * ((values[0][0] * values[1][1]
                                            - values[0][1] * values[1][0])**2) / (expect)

print chisqstat(Ntot, sample_values, expected)

1.58939333193


With a chisqstat of 1.589 and assuming a 0.05 probability, the critical value is equal to 3.84. 

Since 1.589 is smaller than 3.84, we would fail to reject the Null hypothesis.